### PR TITLE
Fix edit command

### LINKED ISSUE #105816: kubectl edit invalid content will remove command interpreter which begin with `#`
### What happened?

1. create configmap, like 

```
kind: ConfigMap
metadata:
  name: test
  namespace: default
apiVersion: v1
data:
  foo.sh: |
    #!/bin/sh
```

2. kube edit  configmap test, and make some error and write-save, will see below

```
# configmaps "test" was not valid:
# * <nil>: Invalid value: "The edited file failed validation": [yaml: line 5: could not find expected ':', invalid character 'a' looking for beginning of value]
#
apiVersion: v1
data:
  foo.sh: |
aaa
```

### What did you expect to happen?

when the content after edited is invalid, should not remove data , thus mean expect content below 

```
# configmaps "test" was not valid:
# * <nil>: Invalid value: "The edited file failed validation": [yaml: line 5: could not find expected ':', invalid character 'a' looking for beginning of value]
#
apiVersion: v1
data:
  foo.sh: |
    #!/bin/sh
aaa

```

### How can we reproduce it (as minimally and precisely as possible)?

1. create configmap

```yaml
kind: ConfigMap
metadata:
  name: test
  namespace: default
apiVersion: v1
data:
  foo.sh: |
    #!/bin/sh
```

2 edit configmap make invalid

### Anything else we need to know?

_No response_

### Kubernetes version

version

```console

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-09-06T02:30:46Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:39:34Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"linux/amd64"}

```

### Cloud provider

kubeadm

```console
kubeadm init --pod-network-cidr=10.10.0.0/16 --service-cidr=10.20.0.0/16 --kubernetes-version=v1.22.1  --cri-socket /run/containerd/containerd.sock
```

### OS version

```console
# On Linux:
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"

$ uname -a
Linux pc 5.11.0-38-generic #42~20.04.1-Ubuntu SMP Tue Sep 28 20:41:07 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by yylt**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #105816

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by lauchokyip — APPROVED** [review]

**Comment by yylt** [comment]
@seans3  hi, could you please review or add some comments ?

File: staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go [hunk 1]
```diff
@@ -682,7 +682,9 @@ func ManualStrip(file []byte) []byte {
 	stripped := []byte{}
 	lines := bytes.Split(file, []byte("\n"))
 	for i, line := range lines {
-		if bytes.HasPrefix(bytes.TrimSpace(line), []byte("#")) {
+		trimline := bytes.TrimSpace(line)
+
+		if bytes.HasPrefix(trimline, []byte("#")) && !bytes.HasPrefix(trimline, []byte("#!")) {
```
[the remark this answers is not in the document]
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go:687]: It is a widely adopted mark, so I think we can go with it.

**Review by soltysh — APPROVED** [review]
/approve
/priority backlog
/triage accepted

**Pull request merged**